### PR TITLE
timeout and state of advertising & keyboard leds info

### PR DESCRIPTION
**Pull request by xiongyihui**
+ add `timeout` to `BLERadio.start_advertising`
+ add `BLERadio.advertising` property
+ add `ReportOut.report` property to get keyboard leds info

### CODE REVIEW

File: adafruit_ble/__init__.py [hunk 1]
```diff
@@ -162,14 +162,16 @@ def __init__(self, adapter=None):
         self._current_advertisement = None
         self._connection_cache = {}
 
-    def start_advertising(self, advertisement, scan_response=None, interval=0.1):
+    def start_advertising(self, advertisement, scan_response=None, interval=0.1, timeout=0):
```
Comment (dhalbert) [inline adafruit_ble/__init__.py:165]: Use `timeout=None` to indicate no timeout. That is more Pythonic, and the way we do it in other similar API's.

File: adafruit_ble/__init__.py [hunk 2]
```diff
@@ -162,14 +162,16 @@ def __init__(self, adapter=None):
...
         """
         Starts advertising the given advertisement.
 
         :param buf scan_response: scan response data packet bytes.
             If ``None``, a default scan response will be generated that includes
             `BLERadio.name` and `BLERadio.tx_power`.
         :param float interval:  advertising interval, in seconds
+        :param int timeout:  advertising timeout in seconds.
+            If 0, no timeout.
```
Comment (dhalbert) [inline adafruit_ble/__init__.py:174]: if `None`, no timeout.

File: adafruit_ble/__init__.py [hunk 3]
```diff
@@ -184,6 +186,7 @@ def start_advertising(self, advertisement, scan_response=None, interval=0.1):
             scan_response=scan_response_bytes,
             connectable=advertisement.connectable,
             interval=interval,
+            timeout=timeout,
```
Comment (dhalbert) [inline adafruit_ble/__init__.py:189]: `timeout=0 if timeout is None else timeout`

File: adafruit_ble/__init__.py [hunk 4]
```diff
@@ -321,3 +324,8 @@ def tx_power(self, value):
     def address_bytes(self):
         """The device address, as a ``bytes()`` object of length 6."""
         return self._adapter.address.address_bytes
+
+    @property
+    def advertising(self):
+        """The adverstising state"""
```
Comment (dhalbert) [inline adafruit_ble/__init__.py:330]: "advertising" (spelling)

File: adafruit_ble/services/standard/hid.py [hunk 5]
```diff
@@ -241,6 +241,10 @@ def __init__(self, service, report_id, usage_page, usage, *, max_length):
             initial_value=struct.pack("<BB", self._report_id, _REPORT_TYPE_OUTPUT),
         )
 
+    @property
```
Comment (dhalbert) [inline adafruit_ble/services/standard/hid.py:244]: Add a docstring.

**Comment by dhalbert** [comment]
Run the "black" formatter over the changed source files, to fix the build failure.

**Comment by xiongyihui** [comment]
Thanks for the review. Updated

File: adafruit_ble/__init__.py [hunk 6]
```diff
@@ -181,6 +183,7 @@ def start_advertising(self, advertisement, scan_response=None, interval=0.1, tim
             scan_response.tx_power = self.tx_power
         if scan_response:
             scan_response_bytes = bytes(scan_response)
+        timeout = 0 if timeout is None else timeout
```
Comment (dhalbert) [inline adafruit_ble/__init__.py:186]: You can remove this from here and put it inside the call to `self._adapter.start_advertising(). Sorry I didn't make that clearer.
```python
        self._adapter.start_advertising(
            advertisement_bytes,
            scan_response=scan_response_bytes,
            connectable=advertisement.connectable,
            interval=interval,
            timeout=0 if timeout is None else timeout,
        )
```

**Comment by dhalbert** [comment]
BTW, you can allow us to make simple edits to this or future PR's if you check a box. See:
https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by xiongyihui** [comment]
Thanks.  Will do that next time.

**Comment by dhalbert** [comment]
Released in https://github.com/adafruit/Adafruit_CircuitPython_BLE/releases/tag/7.2.0.

**Pull request merged**